app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryListScreen.kt

```diff
@@ -385,21 +388,30 @@ fun GalleryListScreen(lub: ListUrlBuilder, navigator: DestinationsNavigator) = c
                     }
                     itemsIndexed(quickSearchList, key = { _, item -> item.id!! }) { index, item ->
                         ReorderableItem(reorderableLazyListState, key = item.id!!) { isDragging ->
-                            val dismissState = rememberSwipeToDismissBoxState(
-                                confirmValueChange = { value ->
+                            // Not using rememberSwipeToDismissBoxState to prevent LazyColumn from reusing it
```

Comment:
Why `SwipeToDismissBoxState` will be reused given `rememberSwipeToDismissBoxState`, but `remember` it manually not?

Reply:
`rememberSwipeToDismissBoxState` uses `rememberSavable` under the hood.

Reply:
Yeah, I mean `rememberSavable` should not return the same item within different lazy items, semanticly, doesn't it?

Reply:
Well, they're different items with the same key (e.g. remove an item then add back it back).

Reply:
> Well, they're different items with the same key (e.g. remove an item then add it back).

Then we should make the key unique

Reply:
The key is unique (for existing items).
Maybe we should use DisposableEffect to reset dismiss states.

Reply:
> Well, they're different items with the same key (e.g. remove an item then add it back).

For which case will we add a item previously removed(have same id), not one newly inserted to dao?

Reply:
If you remove the last item, the newly inserted item will have the same id.

Reply:
Do we really need dismissState to be saved?

Reply:
> Do we really need dismissState to be saved?

We'd better do it otherwise dismissState will not be kept during screen configuration changes. Moreover I'm afraid different items share the same key would cause undefined behaviour.

Reply:
Why would it need to be kept during configuration changes? The existing items' dismissState should always be `Settled`.

Reply:
In case one swiped a item then dialog shown, but he rotated the screen

Reply:
In that case, the dialog will flicker, I'm not sure if it's actually better than just keep it dismissed.

And honestly, who's gonna do that?